### PR TITLE
Feature/available positions designations. TM-695

### PR DESCRIPTION
**Pull request by rtirserio**
New endpoints and models for highlighting available positions as well as the designations (hard to fill, urgent vacancy and volunteer).

### CODE REVIEW

**Comment by mjoyce91** [comment]
Can we join the available positions response with the designation data?

File: talentmap_api/available_positions/views/available_position.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        AvailablePositionFavorite.objects.get_or_create(user=user, cp_id=pk)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def delete(self, request, pk, format=None):
+        '''
+        Removes the available position from favorites
+        '''
+        user = UserProfile.objects.get(user=self.request.user)
+        AvailablePositionFavorite.objects.filter(user=user, cp_id=pk).delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
```
Comment (mjoyce91) [inline talentmap_api/available_positions/views/available_position.py]: I wonder if we'd want to return a 404 if the attempted ID was not in the user's favorites

Comment (mjoyce91) [reply talentmap_api/available_positions/views/available_position.py]: Well, maybe not a 404, but some type of status code or response message

Comment (rtirserio) [reply talentmap_api/available_positions/views/available_position.py]: I was thinking about that when i was setting this up and thought that we don't really care if they try to delete something thats not there.

**Comment by rtirserio** [comment]
@mjoyce91 - added the designations to the returned payload

**Comment by mjoyce91** [comment]
Seeing the same issue as https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/159#issuecomment-533570019

**Comment by rtirserio** [comment]
@mjoyce91 fixed

**Pull request merged**